pkg/cloudprovider/aws/amifamily/resolver.go

```diff
@@ -75,6 +75,7 @@ type AMIFamily interface {
 	DefaultMetadataOptions() *v1alpha1.MetadataOptions
 	EphemeralBlockDevice() *string
 	EphemeralBlockDeviceOverhead() resource.Quantity
+	ENILimitedMemoryOverhead() bool
```

Comment:
I think `MaxPods(i *InstanceType) int` might've been more flexible so if there's further divergence in how different AMIs are calculating MaxPods, we can just modify the AMI specific calculation and leave the others. 

Not blocking on this PR though, we can re-work that later 👍

Reply:
For this specific issue something like `MemoryOverhead(i *InstanceType) resource.Quantity` might be better (as the difference is in the calculation of the overhead, the max pods match correctly in the current implementation). Nonetheless, I think both functions would be a great addition and would give more flexibility in the implementation. 👍